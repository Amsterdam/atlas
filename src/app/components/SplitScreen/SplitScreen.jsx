import React from 'react';
import PropTypes from 'prop-types';

/* istanbul ignore next */ // TODO: refactor, test
const SplitScreen = ({ leftComponent, rightComponent }) => (
  <div className="c-dashboard__column-holder">
    <div
      className="c-dashboard__column u-col-sm--4 qa-dashboard__column--middle
        u-page-break-after"
    >
      {leftComponent}
    </div>
    <div
      className="c-dashboard__column c-dashboard__content u-col-sm--8
        qa-dashboard__column--right"
    >
      {rightComponent}
    </div>
  </div>
);

SplitScreen.propTypes = {
  leftComponent: PropTypes.node.isRequired,
  rightComponent: PropTypes.node.isRequired
};

export default SplitScreen;
