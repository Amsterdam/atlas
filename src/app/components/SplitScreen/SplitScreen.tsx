import { FunctionComponent, ReactNode } from 'react'

export interface SplitScreenProps {
  leftComponent: ReactNode
  rightComponent: ReactNode
  printMode: boolean
}

const SplitScreen: FunctionComponent<SplitScreenProps> = ({
  leftComponent,
  rightComponent,
  printMode,
}) => (
  <div className="c-dashboard__column-holder">
    <div
      className={`
        c-dashboard__column
        u-col-sm--${printMode ? '12' : '4'}
        qa-dashboard__column--middle

      `}
    >
      {leftComponent}
    </div>
    <div
      className={`
        c-dashboard__column
        c-dashboard__content
        u-overflow--y-auto
        u-col-sm--${printMode ? '12' : '8'}
        qa-dashboard__column--right
      `}
    >
      <div style={{ width: '100%' }}>{rightComponent}</div>
    </div>
  </div>
)

export default SplitScreen
