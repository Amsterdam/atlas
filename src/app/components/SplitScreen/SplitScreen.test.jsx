import React from 'react'
import { shallow } from 'enzyme'
import SplitScreen from './SplitScreen'

// Todo: properly test this
describe('SplitScreen', () => {
  it('should render', () => {
    const component = shallow(
      <SplitScreen
        leftComponent={<p>Hello</p>}
        rightComponent={<span>World</span>}
        printMode={false}
      />,
    )
    expect(component).toMatchSnapshot()
  })

  it('should render in print mode', () => {
    const component = shallow(
      <SplitScreen leftComponent={<p>Hello</p>} rightComponent={<span>World</span>} printMode />,
    )
    expect(component).toMatchSnapshot()
  })
})
