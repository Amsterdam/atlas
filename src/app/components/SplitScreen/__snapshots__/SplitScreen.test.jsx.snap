// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitScreen should render 1`] = `
<div
  className="c-dashboard__column-holder"
>
  <div
    className="c-dashboard__column u-col-sm--4 qa-dashboard__column--middle u-page-break-after"
  >
    <p>
      Hello
    </p>
  </div>
  <div
    className="c-dashboard__column c-dashboard__content u-overflow--y-auto u-col-sm--8 qa-dashboard__column--right"
  >
    <span>
      World
    </span>
  </div>
</div>
`;
